src/EnergyPlus/DataBranchNodeConnections.hh

```diff
@@ -148,20 +131,7 @@ struct BranchNodeConnectionsData : BaseGlobalStruct
 
     void clear_state() override
     {
-        this->NumCompSets = 0;
-        this->NumNodeConnectionErrors = 0;
-        this->NumOfNodeConnections = 0;
-        this->MaxNumOfNodeConnections = 0;
-        this->NodeConnectionAlloc = 1000;
```

Comment:
This variable smells funny.

Reply:
It smells funny for two reasons:
1. We are carefully allocating out nodes during initialization in chunks, and I think we can rely on C++ to be smart enough to do this behind-the-scenes reservation.  Anyway, the second reason is that
2. This variable is only used in one function, so I'm moving it to a constexpr int inside that function.